scripts/TrapSpike.gd

```diff
@@ -0,0 +1,44 @@
+extends Area2D
+
+export var time_initial_delay = 0.0
+export var time_disabled = 2.0
+export var time_fadein = 0.5
+export var time_enabled = 0.5
+
+signal trap_triggered
+
+enum TrapSpikeState {
+    Disabled,
+    Fadein,
+    Enabled,
+}
+
+var _state = TrapSpikeState.Disabled
+
+func _ready():
+    add_to_group("traps")
+    _start_trap_cycle()
+
+func _physics_process(delta):
+    if _state == TrapSpikeState.Enabled:
+        for body in self.get_overlapping_bodies():
+            if body.name == "Player":
+                emit_signal("trap_triggered")
+
+func _start_trap_cycle():
+    var tree = get_tree()
+    yield(tree.create_timer(time_initial_delay), "timeout")
+    while true:
```

Comment:
I'd add blank line above this line. I noticed many people don't do it, but I like to have blank lines between blocks of code (eg. before each `if`, `while`, `match` etc; between logical sections like variable definitions, function calls etc). This is how my functions usualy end up looking:

```
func whatever():
    var a = 1
    var b = 2
    var c = 3

    func1()
    func2()

    if a:
        func3()

    func(4)

What do you think?
```

Reply:
I have the same style 😄 , although it may not look like it. Dunno why I didn't apply it here.